### PR TITLE
fix(bulk-import): bulk import fixes

### PR DESCRIPTION
**Pull request by debsmita1**
**Resolves:**
https://issues.redhat.com/browse/RHIDP-3942

**Fixes:**
- Handled invalid entity YAML in the pull request (empty file or entity yaml without `apiVersion` or `kind` or `metadata.name` is considered invalid)
- As of today, the preview panel supports the visualization of only one entity from a catalog-info.yaml file. If the entity YAML file has multiple entities, the preview panel will show only the first entity
- Fixed the `navigate to` url after creating the import
- Removes the repository query param from the import page URL if the repository is not added or the URL/branch is invalid
- Added error handling in the side panel

**Screenshot:**

Screenshot 2024-09-12 at 12 19 51 PM
Screenshot 2024-09-12 at 12 18 57 PM

### CODE REVIEW

**Comment by ShiranHi** [comment]
@debsmita1 I think we can add a space between the two info-bars. 

About the copy of the first one - it will be great if we will improve it, what about: 
The entity YAML in your pull request is invalid (empty file or missing `apiVersion`, `kind`, or `metadata.name`). A new YAML has been generated below. 
Is that accurate? 
I will be happy to discuss this with our UX Writer

**Comment by debsmita1** [comment]
> @debsmita1 I think we can add a space between the two info-bars.
> 
> About the copy of the first one - it will be great if we will improve it, what about: The entity YAML in your pull request is invalid (empty file or missing `apiVersion`, `kind`, or `metadata.name`). A new YAML has been generated below. Is that accurate? I will be happy to discuss this with our UX Writer

Fixed
Screenshot 2024-09-12 at 12 18 57 PM

File: plugins/bulk-import/src/components/PreviewFile/PreviewFileSidebar.tsx [hunk 1]
```diff
@@ -224,52 +234,116 @@ export const PreviewFileSidebar = ({
...
   const classes = useDrawerStyles();
   const bulkImportApi = useApi(bulkImportApiRef);
+  const identityApi = useApi(identityApiRef);
+  const configApi = useApi(configApiRef);
   const [pullRequest, setPullRequest] = React.useState<PullRequestPreviewData>(
     {},
   );
   const [isInitialized, setIsInitialized] = React.useState(false);
 
+  const fetchPullRequestData = async (
```
Comment (invincibleJai) [inline plugins/bulk-import/src/components/PreviewFile/PreviewFileSidebar.tsx:247]: can we abstract this to util and add tests for it

File: plugins/bulk-import/src/components/PreviewFile/PreviewFileSidebar.tsx [hunk 2]
```diff
@@ -224,52 +234,116 @@ export const PreviewFileSidebar = ({
...
-        newPullRequestData[data.id] = data.catalogInfoYaml
-          ?.prTemplate as PullRequestPreview;
-      }
     }
-
     setPullRequest(newPullRequestData);
     setIsInitialized(true);
-  }, [data, bulkImportApi]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data, bulkImportApi, setStatus, status]);
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply plugins/bulk-import/src/components/PreviewFile/PreviewFileSidebar.tsx:162]: moved the drawer component to a separate file

File: plugins/bulk-import/src/components/PreviewFile/PreviewFileSidebar.tsx (hunk 1 above)
Comment (debsmita1) [reply plugins/bulk-import/src/components/PreviewFile/PreviewFileSidebar.tsx:63]: This util has an apiRef which is why I didn't move it from this file

**Review by invincibleJai — APPROVED** [review]
/lgtm

**Comment by ShiranHi** [comment]
After taking with Allison, let's use this text for the info bar: 
The YAML in your pull request is empty or missing `apiVersion`, `kind`, or `metadata.name` . A new YAML has been automatically generated for you.

**Pull request merged**